runtime/vm/exceptiondescribe.c

```diff
@@ -342,11 +425,13 @@ iterateStackTrace(J9VMThread * vmThread, j9object_t* exception, callback_func_t
 				} else {
 					pruneConstructors = FALSE;
 #endif
-					romClass = findROMClassFromPC(vmThread, methodPC, &classLoader);
-					if(romClass) {
-						romMethod = findROMMethodInROMClass(vmThread, romClass, methodPC);
-						if (romMethod != NULL) {
-							methodPC -= (UDATA) J9_BYTECODE_START_FROM_ROM_METHOD(romMethod);
+					if (FALSE == methodPCCacheHit) {
+						romClass = findROMClassFromPC(vmThread, methodPC, &classLoader);
+						if(romClass) {
+							romMethod = findROMMethodInROMClass(vmThread, romClass, methodPC);
+							if (romMethod != NULL) {
+								methodPC -= (UDATA) J9_BYTECODE_START_FROM_ROM_METHOD(romMethod);
+							}
 						}
 					}
```

Comment:
> I think this can be sped in another way. ROMMethods are variable sized making walking them more expensive. For JITTed methods, we have the RAMClass in hand already which would allow walking the `J9Methods` which are fixed sized. This should improve the speed of that operation.

The `findROMMethodInROMClass` which appears on the profile shows only on the interpreted method path seen above. The JIT method path already extracts the `J9ROMMethod` from the `J9Method` structure as seen here:

https://github.com/eclipse/openj9/pull/8053/files#diff-32ef01e828ed7705ed9d726657f20b8dR421-R424

@DanHeidinga is this what you meant? I'm not sure how we can improve this further.

Reply:
The peekClassHashTable call can be moved into the interpreter path so that we will (almost) always have the J9Class available.  We can use that to walk the `J9Methods` rather than the `J9ROMMethods`:
```c
romClass = findROMClassFromPC(vmThread, methodPC, &classLoader);
if (romClass) {
	J9Class *ramClass = peekClassHashTable(....);
	if (ramClass != NULL) {
		const U_32 methodsCount = romClass->romMethodCount;
		J9Method *methods = ramClass->ramMethods;
		for (U_32 i = 0; i < methodsCount; i++) {
			J9ROMMethod *possibleMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methods[i]);
			if ((methodPC >= J9_BYTECODE_START_FROM_ROM_METHOD(possibleMethod))
			&& (methodPC < J9_BYTECODE_END_FROM_ROM_METHOD(possibleMethod)) {
				romMethod = possibleMethod;
				break;
			}
		}
				
	}
}
```
This is roughly what the code would need to do.  It should be faster as `J9Method` is a fixed size structure so we can index it via `methods[i]` rather than iterating the variable sized `J9ROMMethods`.

Final solution will be slightly more complex as it will need to take replaced classes into account.

Reply:
Thanks for describing what you meant in great detail. I've implemented it in [1]. There are performance results from the change below.

[1] https://github.com/fjeremic/openj9/commit/f6d339b4088d5c69c1b3564cc451cf31f6d0ef91

Reply:
One possible improvement is to call `peekClassHashTable` directly rather than indirectly thru the J9InternalVMFunctions table.